src/version.ts

```diff
@@ -0,0 +1,19 @@
+import * as core from '@actions/core';
+import * as fetch from 'node-fetch';
+
+const versionInfo = async (platform: string, version?: string) => {
+  if (version) {
+    core.info(`==> Running version ${version}`);
+  }
+
+  try {
+    const metadataRes = await fetch( `https://uploader.codecov.io/${platform}/latest`, {
+      headers: {'Accept': 'application/json'},
+    });
+    const metadata = await metadataRes.json();
+    core.info(`==> Running version ${metadata['version']}`);
+  } catch (err) {
+    core.info(`Could not pull latest version information: ${err}`);
+  }
+};
+export default versionInfo;
```

Comment:
What is the intention behind this functionality? E.g., is this supposed to print the version that would actually run, the value of the string ultimately used to fetch the uploader, etc.?

Reply:
Intention is to print out the version of the uploader if on `latest`. This will be extremely helpful for debugging (as we only mark `0.1.0`)

Reply:
Understood.

Could we derive this more accurately via the SHA verification somehow? This method of retrieval is not guaranteed to be accurate (the version is pulled *after* the uploader is retrieved—these could be different depending on if/when a release occurs between the requests).

Reply:
Yeah, unfortunately, this is the only API handle we have and it's not at the same time nor the same endpoint. I totally agree with you